core/shared/src/main/scala/magnolia.scala

```diff
@@ -225,7 +225,14 @@ object Magnolia {
       }
 
       val result = if (isCaseObject) {
-        val obj = GlobalUtil.patchedCompanionRef(c)(genericType)
+        val obj = {
```

Comment:
Maybe a simple `val obj = genericType.typeSymbol.name.toTermName` would suffice here?

Reply:
I think `val obj = genericType.typeSymbol.asClass.module` is most precise

Reply:
Ok, great.
I'll go for that then here.
Thank you!